### PR TITLE
LPS-133416 | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-133416
cc @Tim-Cao

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 1]
```diff
@@ -1511,6 +1511,103 @@ definition {
...
+
+		task ("Add a Dynamic Collection based on Document") {
+			JSONAssetlist.addDynamicAssetListEntryWithItemType(
+				groupName = "Test Site Name",
+				itemSubtype = "Document Type Name",
+				itemType = "Document",
+				title = "Dynamic Collection");
+		}
+
+		task ("Enable Filter by Field and Choose Select from List field") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:1558]: Filter by Option 1 from Select from List field

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 2]
```diff
@@ -1511,6 +1511,103 @@ definition {
...
+				siteName = "Test Site Name");
+
+			for (var title : list "1,2") {
+				AssetPublisherPortlet.viewAssetPG(
+					assetContent = "DM Document Description ${n}",
+					assetTitle = "Document Title ${n}");
+			}
+		}
+
+		task ("Select the created dynamic asset list to be used in the asset publisher portlet") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:1592]: Select the dynamic collection in Asset Publisher

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 3]
```diff
@@ -1511,6 +1511,103 @@ definition {
...
+				key_fieldFieldLabel = "Select from List",
+				locator1 = "APConfigurationFilterbyfield#SELECT_FROM_LIST",
+				value1 = "Option 1");
+
+			Button.click(button = "Apply");
+
+			PortletEntry.save();
+		}
+
+		task ("View both Document in Asset Publisher on Widget Page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:1580]: View both documents are shown in Asset Publisher

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 4]
```diff
@@ -1511,6 +1511,103 @@ definition {
...
+			}
+		}
+
+		task ("Select the created dynamic asset list to be used in the asset publisher portlet") {
+			AssetPublisherPortlet.selectAssetList(assetListName = "Dynamic Collection");
+
+			IFrame.closeFrame();
+		}
+
+		task ("View the First Document is shown in Asset Publisher on Widget Page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:1598]: View only the first document is shown in Asset Publisher

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 5]
```diff
@@ -1511,6 +1511,103 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-113996. View the selected collection in Asset Publisher should be shown on the page."
+	@priority = "4"
+	test ViewSelectedCollectionInAssetPublisherOnWidgetPage {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:1516]: FilterDocumentsByField

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 6]
```diff
@@ -1511,6 +1511,103 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-113996. View the selected collection in Asset Publisher should be shown on the page."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:1514]: The user could filter documents by field in Asset Publisher.

**Comment by Tim-Cao** [comment]
Hi @ozysouza, could you update case according to comments?

**Comment by ozysouza** [comment]
> Hi @ozysouza, could you update case according to comments?

Hi @Tim-Cao PR updated following your comments.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 7]
```diff
@@ -793,6 +793,103 @@ definition {
...
+				dmDocumentTypeFieldNames = "Select from List",
+				dmDocumentTypeName = "Document Type Name");
+		}
+
+		task ("Add two options to Select from List") {
+			for (var index : list "1,2") {
+				DataEngine.addFieldOption(
+					fieldFieldLabel = "Select from List",
+					index = "${index}",
+					optionValue = "Option ${index}");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:814]: Add steps to define option value

**Comment by Tim-Cao** [comment]
Hi @ozysouza, see changes https://github.com/liferay-echo/liferay-portal/compare/f31e7763f8edc40f4bf9df712820ff88b8d5aaf3...418adf5fe6c6d878a2987d4cdfa90e4cfd04df63
in https://github.com/liferay-echo/liferay-portal/pull/5036

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**